extractor.go

```diff
@@ -332,10 +353,12 @@ func extract(ctx context.Context, t Target, dst string, src archiveWalker, cfg *
 				// do not end on error
 				continue
 			}
+			if cfg.PreserveFileAttributes() {
+				extractedEntries = append(extractedEntries, ae)
+			}
```

Comment:
I'm not sure I understand this logic in isolation, maybe a comment to explain would be helpful here?

I see two more instances of this pattern below on `400` and `431`, same comment applies to those as well

Reply:
Background to that is that we do not need to consume more memory then necessary for
https://github.com/hashicorp/go-extract/blob/03a7cb637658d37ed89ceb56767a850b2b2d4bb3/extractor.go#L282

Even so, if we do not process them here
https://github.com/hashicorp/go-extract/blob/03a7cb637658d37ed89ceb56767a850b2b2d4bb3/extractor.go#L283-L292

we do not need to store them. Maybe that is not the most elegant way. Happy to optimize on that, but adding all entries to the slice `extractedEntries` for no post processing feels also kind of not necessary. That was the motivation. 

If we look into consideration memory allocation vs. branching/cpu cycles, not-allocating seems to be more performed IMHO. For the sake of simplicity, we can top that branching here.